dataactcore/interfaces/function_bag.py

```diff
@@ -529,7 +406,197 @@ def create_submission(user_id, submission_values, existing_submission_id):
+
+    Arguments:
+    upload_files -- list of named tuples that describe files uploaded to the broker
+    submission -- submission
+    existing_submission -- true if we should update jobs in an existing submission rather than creating new jobs
+
+    Returns:
+    Dictionary of upload ids by filename to return to client, used for calling finalize_submission route
+    """
+    sess = GlobalDB.db().session
+    submission_id = submission.submission_id
+
+    # create the file upload and single-file validation jobs and
+    # set up the dependencies between them
+    jobs_required = []
+    upload_dict= {}
+    for upload_file in upload_files:
+        validation_job_id, upload_job_id = add_jobs_for_file_type(upload_file, submission_id, existing_submission)
+        if validation_job_id:
+            jobs_required.append(validation_job_id)
+        upload_dict[upload_file.file_type] = upload_job_id
+
+    # once single-file upload/validation jobs are created, create the cross-file
+    # validation job and dependencies
+    # todo: remove external validation jobs from the code-base--they aren't used
+    if existing_submission:
+        # find cross-file and external validation jobs and mark them as waiting
+        # (note: job_type of 'validation' is a cross-file job)
+        val_job = sess.query(Job).\
+            filter_by(
+                submission_id = submission_id,
+                job_type_id = JOB_TYPE_DICT["validation"]).\
+            one()
+        val_job.job_status_id = JOB_STATUS_DICT["waiting"]
+        ext_job = sess.query(Job).\
+            filter_by(
+                submission_id = submission_id,
+                job_type_id = JOB_TYPE_DICT["external_validation"]).\
+            one()
+        ext_job.job_status_id = JOB_STATUS_DICT["waiting"]
+        submission.updated_at = time.strftime("%c")
+    else:
+        # create cross-file validation job
+        validation_job = Job(
+            job_status_id=JOB_STATUS_DICT["waiting"],
+            job_type_id=JOB_TYPE_DICT["validation"],
+            submission_id=submission_id)
+        sess.add(validation_job)
+        # create external validation job
+        external_job = Job(
+            job_status_id=JOB_STATUS_DICT["waiting"],
+            job_type_id=JOB_TYPE_DICT["external_validation"],
+            submission_id=submission_id)
+        sess.add(external_job)
+        sess.flush()
+        # create dependencies for validation jobs
+        for job_id in jobs_required:
+            val_dependency = JobDependency(job_id=validation_job.job_id, prerequisite_id=job_id)
+            sess.add(val_dependency)
+            ext_dependency = JobDependency(job_id=external_job.job_id, prerequisite_id=job_id)
+            sess.add(ext_dependency)
+
+    sess.commit()
+    upload_dict["submission_id"] = submission_id
+    return upload_dict
+
+def add_jobs_for_file_type(upload_file, submission_id, existing_submission):
```

Comment:
No biggie, but the `for_file_type` here doesn't seem to make sense anymore. `for_uploaded_file`? Also, this is outside the scope for this PR, but this function is waaaaay too big. We should break it up.